crates/miden-tx/src/tests/kernel_tests/test_tx.rs

```diff
@@ -1077,35 +1291,41 @@ fn test_fpi_execute_foreign_procedure() {
 // HELPER FUNCTIONS
 // ================================================================================================
 
-fn get_mock_fpi_adv_inputs(foreign_account: &Account, mock_chain: &MockChain) -> AdviceInputs {
+fn get_mock_fpi_adv_inputs(
+    foreign_accounts: Vec<&Account>,
+    mock_chain: &MockChain,
+) -> AdviceInputs {
     let mut advice_inputs = AdviceInputs::default();
-    TransactionKernel::extend_advice_inputs_for_account(
-        &mut advice_inputs,
-        &foreign_account.clone().into(),
-        foreign_account.code(),
-        &foreign_account.storage().get_header(),
-        // Provide the merkle path of the foreign account to be able to verify that the account
-        // database has the hash of this foreign account. Verification is done during the
-        // execution of the `kernel::account::validate_current_foreign_account` procedure.
-        &MerklePath::new(
-            mock_chain
-                .accounts()
-                  // TODO: Update.
-                .open(&LeafIndex::<ACCOUNT_TREE_DEPTH>::new(foreign_account.id().prefix().as_felt().as_int()).unwrap())
-                .path
-                .into(),
-        ),
-    )
-    .unwrap();
 
-    for slot in foreign_account.storage().slots() {
-        // if there are storage maps, we populate the merkle store and advice map
-        if let StorageSlot::Map(map) = slot {
-            // extend the merkle store and map with the storage maps
-            advice_inputs.extend_merkle_store(map.inner_nodes());
-            // populate advice map with Sparse Merkle Tree leaf nodes
-            advice_inputs
-                .extend_map(map.leaves().map(|(_, leaf)| (leaf.hash(), leaf.to_elements())));
+    for foreign_account in foreign_accounts {
+        TransactionKernel::extend_advice_inputs_for_account(
+            &mut advice_inputs,
+            &foreign_account.into(),
+            foreign_account.code(),
+            &foreign_account.storage().get_header(),
+            // Provide the merkle path of the foreign account to be able to verify that the account
+            // database has the hash of this foreign account. Verification is done during the
```

Comment:
```suggestion
            // Provide the merkle path of the foreign account to be able to verify that the account
            // tree has the commitment of this foreign account. Verification is done during the
```

nit: use commitment over hash and tree rather than db?